Classes/Service/AnonymizationService.php

```diff
@@ -91,6 +92,22 @@ protected function getAnonymizedPropertyValuesFor($className)
         return $anonymizedPropertyValues;
     }
 
+    /**
+     * Check whether a class is properly annotated as AnonymizableEntity and has anonymizable properties
+     *
+     * @param string $className
+     * @return bool
+     */
+    protected function isAnonymizable($className)
+    {
+        /** @var AnonymizableEntity $annotation */
+        $annotation = $this->reflectionService->getClassAnnotation($className, AnonymizableEntity::class);
```

Comment:
I would prefer to use in_array($className, $this->getAnonymizableClassNames) here (and keep getAnonymizableClassNames as is); this prevents using the reflection service twice.

Reply:
I reverted the changes regarding this method, however this may lead to unfortunate overhead when anonymizing single entities. It is probably advisable that the data currently accumulated to determine anonymizablity in the process is cached in some way.